### PR TITLE
Use lambdas when creating Threads

### PR DESCRIPTION
**Pull request by mikewalch**

### CODE REVIEW

File: server/tracer/src/test/java/org/apache/accumulo/tracer/TracerTest.java [hunk 1]
```diff
@@ -168,12 +168,7 @@ public void testThrift() throws Exception {
     TThreadPoolServer.Args args = new TThreadPoolServer.Args(transport);
     args.processor(new Processor<Iface>(TraceWrap.service(new Service())));
     final TServer tserver = new TThreadPoolServer(args);
-    Thread t = new Thread() {
-      @Override
-      public void run() {
-        tserver.serve();
-      }
-    };
+    Thread t = new Thread(() -> tserver.serve());
```
Comment (ctubbsii) [inline server/tracer/src/test/java/org/apache/accumulo/tracer/TracerTest.java]: Can you just do `tserver::serve`?

File: test/src/main/java/org/apache/accumulo/test/ShellServerIT.java [hunk 2]
```diff
@@ -1570,16 +1570,13 @@ public void listscans() throws Exception {
...
-        public void run() {
-          try {
-            Iterators.size(s.iterator());
-          } catch (Exception ex) {
-            throw new RuntimeException(ex);
-          }
+      Thread thread = new Thread(() -> {
+        try {
+          Iterators.size(s.iterator());
+        } catch (Exception ex) {
```
Comment (ctubbsii) [inline test/src/main/java/org/apache/accumulo/test/ShellServerIT.java]: Does this even have checked exceptions? If so, a multi-catch block would be better than a wide supertype.

File: test/src/main/java/org/apache/accumulo/test/ShellServerIT.java [hunk 3]
```diff
@@ -1570,16 +1570,13 @@ public void listscans() throws Exception {
...
+          throw new RuntimeException(ex);
```
Comment (ctubbsii) [inline test/src/main/java/org/apache/accumulo/test/ShellServerIT.java]: Does this propagate to the parent thread? Or is it lost? If lost, can we fix that?

File: test/src/main/java/org/apache/accumulo/test/SplitCancelsMajCIT.java [hunk 4]
```diff
@@ -76,16 +76,13 @@ public void test() throws Exception {
...
-          try {
-            c.tableOperations().compact(tableName, null, null, true, true);
-          } catch (Exception e) {
-            ex.set(e);
-          }
+      Thread thread = new Thread(() -> {
+        try {
+          c.tableOperations().compact(tableName, null, null, true, true);
+        } catch (Exception e) {
+          ex.set(e);
```
Comment (ctubbsii) [inline test/src/main/java/org/apache/accumulo/test/SplitCancelsMajCIT.java]: Clever way to track exceptions inside the thread. Do we want to do this elsewhere?

File: server/tracer/src/test/java/org/apache/accumulo/tracer/TracerTest.java (hunk 1 above)
Comment (mikewalch) [reply server/tracer/src/test/java/org/apache/accumulo/tracer/TracerTest.java]: Yes I can.

File: test/src/main/java/org/apache/accumulo/test/ShellServerIT.java (hunk 2 above)
Comment (mikewalch) [reply test/src/main/java/org/apache/accumulo/test/ShellServerIT.java]: It doesn't even have checked exceptions. I will fix.

File: test/src/main/java/org/apache/accumulo/test/ShellServerIT.java (hunk 3 above)
Comment (mikewalch) [reply test/src/main/java/org/apache/accumulo/test/ShellServerIT.java]: No checked exceptions were being thrown.

File: test/src/main/java/org/apache/accumulo/test/SplitCancelsMajCIT.java (hunk 4 above)
Comment (mikewalch) [reply test/src/main/java/org/apache/accumulo/test/SplitCancelsMajCIT.java]: I looked around and didn't see anywhere else where this should be used.

**Pull request merged**